### PR TITLE
[terra-form-radio] Fix for RadioField title being read twice on Voice Over 

### PR DESCRIPTION
**Pull request by gt106551**
Summary
Terra - Form Radio Field - VO announces group and legend multiple times

Solution
I have replaced legend Tag with div Tag in Safari & Edge.

Testing
I have tested with Chrome, Safari,Edge.

### CODE REVIEW

File: packages/terra-form-radio/src/RadioField.jsx [hunk 1]
```diff
@@ -114,24 +114,38 @@ const RadioField = (props) => {
...
-            <span className={cx('optional')}>{intl.formatMessage({ id: 'Terra.form.field.optional' })}</span>
-          )}
-        {!isInvalid && <span className={cx('error-icon-hidden')} />}
-      </div>
+      {!isInvalid && <span className={cx('error-icon-hidden')} />}
+    </div>
+  );
+
+  const legendGroup = isSafariOrEdgBrowser ? (
+    <div id={legendAriaDescriptionId} className={cx(['legend-group', { 'legend-group-hidden': isLegendHidden }])}>
```
Comment (supreethmr) [inline packages/terra-form-radio/src/RadioField.jsx:143]: what happens if you keep `div` instead of `legend` for the browsers other than safari and edge...??

Comment (gt106551) [reply packages/terra-form-radio/src/RadioField.jsx:143]: if we replace legend with div in safari and edge is working fine (legend read only once) 
if we replace legend with div in chrome browser legend is not announcing.

File: packages/terra-form-radio/src/RadioField.jsx [hunk 2]
```diff
@@ -114,24 +114,38 @@ const RadioField = (props) => {
...
+      {legendCode}
+    </div>
+  ) : (
+    <legend id={legendAriaDescriptionId} className={cx(['legend-group', { 'legend-group-hidden': isLegendHidden }])}>
+      {legendCode}
```
Comment (supreethmr) [inline packages/terra-form-radio/src/RadioField.jsx:148]:
```suggestion
      {legendCode}
    </div>
  ) : (
    <legend id={legendAriaDescriptionId} className={cx(['legend-group', { 'legend-group-hidden': isLegendHidden }])}>
      {legendCode}
```

File: packages/terra-form-radio/src/RadioField.jsx [hunk 3]
```diff
@@ -114,24 +114,38 @@ const RadioField = (props) => {
...
+        <React.Fragment>
+          <div aria-hidden="true" className={cx('required')}>*</div>
+          <VisualyHiddenText text={intl.formatMessage({ id: 'Terra.form.field.required' })} />
+        </React.Fragment>
+      )}
+      {legend}
+      {required && !isInvalid && hideRequired && <span className={cx('required-hidden')}>*</span>}
+      {showOptional && !required
+        && (
+          <span className={cx('optional')}>{intl.formatMessage({ id: 'Terra.form.field.optional' })}</span>
```
Comment (supreethmr) [inline packages/terra-form-radio/src/RadioField.jsx:136]:
```suggestion
  const Component = (isSafariOREdge) ? 'div' : 'legend';
  const legendGroup = (
    <Component id={legendAriaDescriptionId} className={cx(['legend-group', { 'legend-group-hidden': isLegendHidden }])}>
      <div {...legendAttrs} className={legendClassNames}>
        {isInvalid && <span className={cx('error-icon')} />}
        {required && (isInvalid || !hideRequired) && (
          <React.Fragment>
            <div aria-hidden="true" className={cx('required')}>*</div>
            <VisualyHiddenText text={intl.formatMessage({ id: 'Terra.form.field.required' })} />
          </React.Fragment>
        )}
        {legend}
        {required && !isInvalid && hideRequired && <span className={cx('required-hidden')}>*</span>}
        {showOptional && !required
          && (
            <span className={cx('optional')}>{intl.formatMessage({ id: 'Terra.form.field.optional' })}</span>
          )}
        {!isInvalid && <span className={cx('error-icon-hidden')} />}
      </div>
    </Component>
```

File: packages/terra-form-radio/src/RadioField.jsx [hunk 4]
```diff
@@ -114,8 +114,14 @@ const RadioField = (props) => {
   const errorAriaDescriptionId = error ? `terra-radio-field-description-error-${uniqueid()}` : '';
   const ariaDescriptionIds = `${legendAriaDescriptionId} ${errorAriaDescriptionId} ${helpAriaDescriptionId}`;
 
+  let isSafariOrEdgBrowser = false;
+  if (navigator.userAgent.indexOf('Safari') !== -1 && (navigator.userAgent.indexOf('Chrome') === -1 || navigator.userAgent.indexOf('Edg') !== -1)) {
+    isSafariOrEdgBrowser = true;
+  }
```
Comment (supreethmr) [inline packages/terra-form-radio/src/RadioField.jsx:120]:
```suggestion
  const isSafariOREdge = ((navigator.userAgent.indexOf('Safari') !== -1 || navigator.userAgent.indexOf('Edg') !== -1) && navigator.userAgent.indexOf('Chrome') === -1);
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 5]
```diff
@@ -8,12 +8,23 @@ import Radio from '../../src/Radio';
...
+  userAgentGetter = jest.spyOn(window.navigator, 'userAgent', 'get');
+});
+
 it('should render a default radio field', () => {
   const radioField = (<RadioField legend="Default RadioField" />);
   const wrapper = shallowWithIntl(radioField);
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render radio field with div element or legend element', () => {
```
Comment (supreethmr) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:22]:
```suggestion
it('should render radio field with div element for safari browser', () => {
```

File: packages/terra-form-radio/tests/jest/__snapshots__/RadioField.test.jsx.snap [hunk 6]
```diff
@@ -576,6 +576,27 @@ exports[`should render an optional part on the label 1`] = `
 />
 `;
 
+exports[`should render radio field with div element or legend element 1`] = `
+<fieldset
+  className="radio-field"
+  required={false}
+>
+  <legend
```
Comment (supreethmr) [inline packages/terra-form-radio/tests/jest/__snapshots__/RadioField.test.jsx.snap:584]: since mock value is set to return Safari. it should render div not legend !!

File: packages/terra-form-radio/src/RadioField.jsx (hunk 4 above)
Comment (gt106551) [reply packages/terra-form-radio/src/RadioField.jsx:120]: This condtion working fine in chrome and safari,but inEdge browser ( navigator.userAgent.indexOf('Chrome') this value is 81. navigator.userAgent.indexOf('Chrome') === -1); this condition always false in Edge browser.

File: packages/terra-form-radio/src/RadioField.jsx [hunk 7]
```diff
@@ -114,8 +114,14 @@ const RadioField = (props) => {
...
+
+  const Component = (isSafariOrEdgBrowser) ? 'div' : 'legend';
```
Comment (supreethmr) [inline packages/terra-form-radio/src/RadioField.jsx:122]:
```suggestion
  const Component = (isSafariOREdge) ? 'div' : 'legend';
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 5 above)
Comment (supreethmr) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:22]: we need to add similar tests for edge and chrome browsers also

Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: i have made changes as per suggestions.

File: packages/terra-form-radio/src/RadioField.jsx (hunk 7 above)
Comment (gt106551) [reply packages/terra-form-radio/src/RadioField.jsx]: i have made changes as per suggestions.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 8]
```diff
@@ -8,12 +8,29 @@ import Radio from '../../src/Radio';
...
+});
+
 it('should render a default radio field', () => {
   const radioField = (<RadioField legend="Default RadioField" />);
   const wrapper = shallowWithIntl(radioField);
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render radio field with div element for safari browser or Edg browser', () => {
+  userAgentGetter.mockReturnValue('safari Edg');
```
[the remark this answers is not in the document]
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:23]: i have made changes as per suggestions.

[the remark this answers is not in the document]
Comment (supreethmr) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:23]: I still see `userAgentGetter.mockReturnValue('safari Edg');` in test file can you help me to find the changes made ??

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 9]
```diff
@@ -8,12 +8,29 @@ import Radio from '../../src/Radio';
...
+  const wrapper = shallowWithIntl(<RadioField legend="Coustom Message RadioField" />);
+  expect(wrapper.dive()).toMatchSnapshot();
+});
+
+it('should render radio field with legend element for Chrome browser', () => {
+  userAgentGetter.mockReturnValue('Chrome');
+  const wrapper = shallowWithIntl(<RadioField legend="Coustom Message RadioField" />);
+  expect(wrapper.dive()).toMatchSnapshot();
```
Comment (supreethmr) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:31]: As @sdadn  mentioned on Checkbox Field PR we can have expect() to ensure component is rendered with `legend` element. Probably we can also use [toMatchInlineSnapshots()](https://jestjs.io/docs/snapshot-testing#inline-snapshots) which ensures component is rendered with expected dom elements

Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:31]: I am using .dive() method, it will executes as we expected dom elements.like div or legend based on mockreturn value.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 10]
```diff
@@ -8,12 +8,67 @@ import Radio from '../../src/Radio';
...
+
 it('should render a default radio field', () => {
   const radioField = (<RadioField legend="Default RadioField" />);
   const wrapper = shallowWithIntl(radioField);
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render radio field with div element for Safari browser or Edg browser', () => {
+  userAgentGetter.mockReturnValue('Safari Edg');
+  const wrapper = shallowWithIntl(<RadioField legend="Coustom Message RadioField" />);
```
Comment (supreethmr) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:24]:
```suggestion
  const wrapper = shallowWithIntl(<RadioField legend="Custom Message RadioField" />);
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 11]
```diff
@@ -8,12 +8,67 @@ import Radio from '../../src/Radio';
...
+          />
+        </div>
+      </div>
+    </fieldset>
+  `);
+});
+
+it('should render radio field with legend element for Chrome browser', () => {
+  userAgentGetter.mockReturnValue('Chrome');
+  const wrapper = shallowWithIntl(<RadioField legend="Coustom Message RadioField" />);
```
Comment (supreethmr) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:49]:
```suggestion
  const wrapper = shallowWithIntl(<RadioField legend="Custom Message RadioField" />);
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 12]
```diff
@@ -8,12 +8,67 @@ import Radio from '../../src/Radio';
...
+      required={false}
+    >
+      <legend
+        className="legend-group"
+        id="terra-radio-field-description-2"
+      >
+        <div
+          className="legend"
+        >
+          Coustom Message RadioField
```
Comment (supreethmr) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:62]:
```suggestion
          Custom Message RadioField
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 13]
```diff
@@ -8,12 +8,67 @@ import Radio from '../../src/Radio';
...
+      required={false}
+    >
+      <div
+        className="legend-group"
+        id="terra-radio-field-description-1"
+      >
+        <div
+          className="legend"
+        >
+          Coustom Message RadioField
```
Comment (supreethmr) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:37]:
```suggestion
          Custom Message RadioField
```

**Review by supreethmr — APPROVED** [review]

File: packages/terra-form-radio/src/RadioField.jsx [hunk 14]
```diff
@@ -109,13 +113,17 @@ const RadioField = (props) => {
     legendAttrs.className,
   ]);
 
+  const feildSetId = `terra-radio-group-${uniqueid()}`;
```
Comment (sdadn) [inline packages/terra-form-radio/src/RadioField.jsx:116]: nit: spelling. You may need to fix all instances of this variable.
```suggestion
  const fieldSetId = `terra-radio-group-${uniqueid()}`;
```

File: packages/terra-form-radio/src/RadioField.jsx [hunk 15]
```diff
@@ -109,13 +113,17 @@ const RadioField = (props) => {
...
   const legendAriaDescriptionId = `terra-radio-field-description-${uniqueid()}`;
   const helpAriaDescriptionId = help ? `terra-radio-field-description-help-${uniqueid()}` : '';
   const errorAriaDescriptionId = error ? `terra-radio-field-description-error-${uniqueid()}` : '';
   const ariaDescriptionIds = `${legendAriaDescriptionId} ${errorAriaDescriptionId} ${helpAriaDescriptionId}`;
 
+  const isSafariOREdge = ((navigator.userAgent.indexOf('Safari') !== -1 && navigator.userAgent.indexOf('Chrome') === -1) || navigator.userAgent.indexOf('Edg') !== -1);
```
Comment (sdadn) [inline packages/terra-form-radio/src/RadioField.jsx:122]: Why is the condition for Chrome needed? It is redundant to check for it when both this condition and checking for safari will always return true. If the Safari condition is false, then the chrome check will never be evaluated due to short circuit evaluation.

File: packages/terra-form-radio/src/RadioField.jsx [hunk 16]
```diff
@@ -109,13 +113,17 @@ const RadioField = (props) => {
...
+  const Component = (isSafariOREdge) ? 'div' : 'legend';
```
Comment (sdadn) [inline packages/terra-form-radio/src/RadioField.jsx:123]: `Component` is too vague of a variable name; can we name it to something else like `LegendGroup`?
```suggestion
  const Component = (isSafariOREdge) ? 'div' : 'legend';
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 17]
```diff
@@ -1,70 +1,163 @@
 import React from 'react';
 import ThemeContextProvider from 'terra-theme-context/lib/ThemeContextProvider';
+import uniqueid from 'lodash.uniqueid';
```
Comment (sdadn) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:3]: is there a reason why we're using `lodash.uniqueid`? We typically use `uuid` instead.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 18]
```diff
@@ -1,70 +1,163 @@
...
+          className="legend"
+        >
+          Custom Message RadioField
+          <span
+            className="error-icon-hidden"
+          />
+        </div>
+      </div>
+    </fieldset>
+  `);
```
Comment (sdadn) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:50]: Is there a reason this is being used instead of `toMatchSnapshot`?

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 19]
```diff
@@ -1,70 +1,163 @@
...
+});
+
+it('should render radio field with legend element for Chrome browser', () => {
+  userAgentGetter.mockReturnValue('Chrome');
+  const wrapper = shallowWithIntl(
+    <RadioField legend="Custom Message RadioField" />,
+  );
```
Comment (sdadn) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:57]: Nit: this can be on a single line.
```suggestion
  const wrapper = shallowWithIntl(<RadioField legend="Custom Message RadioField" />);
```

**Review by sdadn — CHANGES_REQUESTED** [review]

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 17 above)
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:3]: It's already installed in that component, that's the reason, I'm using it.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 18 above)
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:50]: As per @supreethmr  suggestion I'm using .toMatchInlineSnapshot().

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 19 above)
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: i have made changes as per suggestions

File: packages/terra-form-radio/src/RadioField.jsx (hunk 16 above)
Comment (gt106551) [reply packages/terra-form-radio/src/RadioField.jsx]: i have made changes as per suggestions

File: packages/terra-form-radio/src/RadioField.jsx (hunk 14 above)
Comment (gt106551) [reply packages/terra-form-radio/src/RadioField.jsx]: i have made changes as per suggestions

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 12 above)
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: i have made changes as per suggestions

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 13 above)
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: i have made changes as per suggestions

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 11 above)
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: i have made changes as per suggestions

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 17 above)
Comment (sdadn) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:3]: I'm confused, won't all consumer tests need to be updated to mock `lodash.uniqueid` as well?

File: packages/terra-form-radio/src/RadioField.jsx (hunk 15 above)
Comment (sycombs) [reply packages/terra-form-radio/src/RadioField.jsx:122]: Can we add a comment since it is not easy to understand just from the code why the Chrome check is necessary?

Also, can we split the `isSafariOREdge` variable into two variables for better readability since it can have multiple implications? Then on [line 123](https://github.com/cerner/terra-core/pull/3804/files#diff-de27e9ca32868da5500732f9a44baeb41aa7159e7a4bd287c0149be277869b2dR123), use `isSafari || isEdge` in the ternary condition.

Something like:

```suggestion
  // The Chrome check is necessary to filter out Chrome browsers on MacOS
  const isSafari = navigator.userAgent.indexOf('Safari') !== -1 && navigator.userAgent.indexOf('Chrome') === -1;
  
  const isEdge = navigator.userAgent.indexOf('Edg') !== -1;
```

**Review by sycombs — CHANGES_REQUESTED** [review]

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 18 above)
Comment (sycombs) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:48]: According to the [Jest docs](https://jestjs.io/docs/snapshot-testing#inline-snapshots),

> Inline snapshots behave identically to external snapshots (.snap files), except the snapshot values are written automatically back into the source code.

Since it seems like we'll be able to achieve the same result with either function, let's use `.toMatchSnapshot` for consistency with other Terra components. We can also specifically test for properties (see [`terra-alert`](https://github.com/cerner/terra-core/blob/main/packages/terra-alert/tests/jest/Alert.test.jsx#L29-L40) for an example.)

Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: i have made changes as per suggestions.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 17 above)
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:3]: consumers of radio button are already using mock for lodash.uniqueid as it was there from beginning and not added as part of this PR"

Comment (sdadn) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:3]: Sorry, looks like I missed where it was already imported in the source. That answers my confusion, thanks for clarifying!

Comment (sdadn) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:3]: Since we're mocking, importing this is unneeded.
```suggestion
```

**Review by sdadn — DISMISSED** [review]
One comment but overall it looks good.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 17 above)
Comment (sdadn) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx:3]: Oh okay, that makes sense.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 20]
```diff
@@ -1,70 +1,119 @@
...
 import RadioField from '../../src/RadioField';
 import Radio from '../../src/Radio';
 
+jest.mock('lodash.uniqueid');
 window.matchMedia = () => ({ matches: true });
 
+let userAgentGetter;
+beforeEach(() => {
+  userAgentGetter = jest.spyOn(window.navigator, 'userAgent', 'get');
+  uniqueid.mockImplementation(() => 'uuid123');
```
Comment (sdadn) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:16]: We don't want to want to set the mock before every test, just once. This can be removed from here and added before the tests globally.
```suggestion
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 21]
```diff
@@ -1,70 +1,119 @@
 import React from 'react';
 import ThemeContextProvider from 'terra-theme-context/lib/ThemeContextProvider';
+import uniqueid from 'lodash.uniqueid';
 
 /* eslint-disable-next-line import/no-extraneous-dependencies */
 import { shallowWithIntl, mountWithIntl } from 'terra-enzyme-intl';
 import RadioField from '../../src/RadioField';
 import Radio from '../../src/Radio';
 
+jest.mock('lodash.uniqueid');
```
Comment (sdadn) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:10]: We cam define the mock once here globally:
```suggestion
jest.mock('lodash.uniqueid');
uniqueid.mockReturnValue('uuid123');
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 22]
```diff
@@ -1,70 +1,119 @@
...
+  const radioField = <RadioField legend="Default RadioField" />;
   const wrapper = shallowWithIntl(radioField);
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render radio field with div element for Safari browser or Edg browser', () => {
+  userAgentGetter.mockReturnValue('Safari Edg');
+  const wrapper = shallowWithIntl(<RadioField legend="Custom Message RadioField" />);
+  expect(wrapper.dive()).toMatchSnapshot();
+});
```
Comment (sdadn) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:29]: Can we add explicit testing for the div element? similar to:
 https://github.com/cerner/terra-core/blob/0c7a90d354e215273f20d9a8841e82a44c3cef9c/packages/terra-alert/tests/jest/Alert.test.jsx#L23-L26

Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: i have made changes as per suggestions.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 21 above)
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: we need to import uniqueid, otherwise it'll throw an error.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 23]
```diff
@@ -1,70 +1,124 @@
...
 });
 
+it('should render radio field with div element for Safari browser or Edg browser', () => {
+  userAgentGetter.mockReturnValue('Safari Edg');
+  const wrapper = shallowWithIntl(<RadioField legend="Custom Message RadioField" />).dive();
+  const radioDiv = wrapper.find('fieldset');
+  const divExist = radioDiv.find('div');
+  expect(divExist).toBeDefined();
+  expect(wrapper).toMatchSnapshot();
+});
```
Comment (sdadn) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:31]: +1 on the test cases

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 24]
```diff
@@ -1,70 +1,124 @@
...
 /* eslint-disable-next-line import/no-extraneous-dependencies */
 import { shallowWithIntl, mountWithIntl } from 'terra-enzyme-intl';
 import RadioField from '../../src/RadioField';
 import Radio from '../../src/Radio';
 
 window.matchMedia = () => ({ matches: true });
+jest.mock('lodash.uniqueid');
+
+let userAgentGetter;
+beforeEach(() => {
```
Comment (sdadn) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx:13]: We do not want to run these for every test
```suggestion
beforeAll(() => {
```

Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: i have made changes as per suggestions.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 20 above)
Comment (gt106551) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: i have made changes as per suggestions.

**Review by sdadn — APPROVED** [review]

**Review by sycombs — APPROVED** [review]
